lib/templateFactory.js

```diff
@@ -187,6 +187,60 @@ class TemplateFactory extends BaseFactory {
         return super.list(config);
     }
 
+    /**
+     * List templates and associated metrics with pagination and filter options
+     * @method listWithMetrics
+     * @param  {Object}   config                  Config object
+     * @param  {Object}   config.params           Parameters to filter on
+     * @param  {Object}   config.paginate         Pagination parameters
+     * @param  {Number}   config.paginate.count   Number of items per page
+     * @param  {Number}   config.paginate.page    Specific page of the set to return
+     * @return {Promise}
+     */
+    listWithMetrics(config) {
+        return this.list(config).then((templates) => {
+            if (templates.length === 0) {
+                return templates;
+            }
+
+            // eslint-disable-next-line global-require
+            const JobFactory = require('./jobFactory');
+            const jobFactory = JobFactory.getInstance();
+
+            // eslint-disable-next-line global-require
+            const BuildFactory = require('./buildFactory');
+            const buildFactory = BuildFactory.getInstance();
+
+            const templateIds = templates.map(t => t.id);
+
+            const listConfig = {
+                params: { templateId: templateIds },
+                readOnly: true
+            };
+
+            return Promise.all([
+                jobFactory.list(listConfig),
+                buildFactory.list(listConfig)]
+            ).then(([jobs, builds]) =>
+                templates.map((t) => {
+                    const numJobs = jobs.filter(j => j.templateId === t.id).length;
```

Comment:
why do we need this check ? Also if we are looking for just count, can't we leverage sequalize/DB to issue a count rather than doing this in API ?

Reply:
Some templates have 50+ versions, so rather than doing 50 count queries, I query for jobs where the templateId is in a list of templateIds (ids for the different template versions) and then filter the list of jobs to get the appropriate counts for each individual template version.

Reply:
Updated to do an aggregation query, should significantly reduce API computation.